### PR TITLE
Document Data.MonoTraversable

### PR DESCRIPTION
**Pull request by kbillings**

### CODE REVIEW

File: src/Data/MonoTraversable.hs [hunk 1]
```diff
@@ -5,19 +5,19 @@
...
 -- All of the laws for the polymorphic typeclasses apply to their monomorphic
--- cousins. Thus, even though a @MonoFunctor@ instance for @Set@ could
+-- cousins. Thus, even though a 'MonoFunctor' instance for 'Set' could
 -- theoretically be defined, it is omitted since it could violate the functor
--- law of @omap f . omap g = omap (f . g)@.
+-- law of @'omap' f . 'omap' g = 'omap' (f . g)@.
 --
 -- Note that all typeclasses have been prefixed with @Mono@, and functions have
--- been prefixed with @o@. The mnemonic for @o@ is \"only one,\" or alternatively
+-- been prefixed with @o@. The mnemonic for @o@ is \"only one\", or alternatively
```
Comment (MaxGabriel) [inline src/Data/MonoTraversable.hs]: I can't find any specific documentation on not needing to escape double quotes, but just based on testing I don't think you need to escape them if they're not surrounding a valid Haskell identifier.

File: src/Data/MonoTraversable.hs [hunk 2]
```diff
@@ -206,112 +209,165 @@ instance VS.Storable a => MonoFunctor (VS.Vector a) where
...
     ocompareLength c0 i0 = olength c0 `compare` fromIntegral i0 -- FIXME more efficient implementation
     {-# INLINE ocompareLength #-}
 
+    -- | Map each element of a monomorphic container to an action,
+    -- evaluate these actions from left to right, and ignore the results.
     otraverse_ :: (MonoFoldable mono, Applicative f) => (Element mono -> f b) -> mono -> f ()
     otraverse_ f = ofoldr ((*>) . f) (pure ())
     {-# INLINE otraverse_ #-}
 
+    -- | 'ofor_' is 'otraverse_' with it's arguments flipped.
```
Comment (MaxGabriel) [inline src/Data/MonoTraversable.hs]: Think this should be "its" not "it's"

**Comment by MaxGabriel** [comment]
This documentation is really good imo. I especially liked the detailed descriptions of the map/fold/traverse functions, the min/maxBy descriptions, and the consistent use of the phrase "monomorphic container". Your descriptions would be a big improvement to the corresponding functions in Data.MinLen as well, if you're interested in cross applying them or letting me steal them.

**Comment by kbillings** [comment]
@MaxGabriel I'm currently working on writing documentation for the other modules of this package as well. I'll make more pull requests when I finish the other modules.

**Comment by gregwebs** [comment]
looking forward to it!

**Pull request merged**